RNTester/RNTesterUnitTests/RCTConvert_UIColorTests.m

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+#import <XCTest/XCTest.h>
+
+#import <React/RCTConvert.h>
+
+@interface RCTConvert_NSColorTests : XCTestCase
+
+@end
+
+@implementation RCTConvert_NSColorTests
+
+- (void)testColor
+{
+  id json = RCTJSONParse(@"{ \"semantic\": \"lightTextColor\" }", nil);
+  UIColor *value = [RCTConvert UIColor:json];
+  XCTAssertEqualObjects(value, [UIColor lightTextColor]);
```

Comment:
interesting how this test passes, it looks like `lightTextColor` is using a `dispatch_one` so you get back the same reference.

Reply:
As it should?
```
(lldb) p value
(UICachedDeviceWhiteColor *) $1 = 0x00007fd0b15056c0
(lldb) p [UIColor lightTextColor]
(UICachedDeviceWhiteColor *) $2 = 0x00007fd0b15056c0
```

Reply:
the fact that these objects are the same is because of an implementation detail that they return the same reference. I think the right approach would be to compare them with RCTGetRGBAColorComponents to evaluate if the color is the same.

Reply:
I am asserting that they are the same reference.   The test is trying to simulate the case of using some future OS semantic color that doesn't exist on the platform being tested so its falling back to an older semantic color.

But I'm happy to change the assert to something like:
`  XCTAssertTrue(CGColorsAreEqual([value CGColor], [[UIColor blueColor] CGColor]));`